orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java

```diff
@@ -132,26 +134,75 @@ public OrchestrationChatResponse chatCompletion(
   @Nonnull
   public CompletionPostResponse executeRequest(@Nonnull final CompletionPostRequest request)
       throws OrchestrationClientException {
-    final BasicClassicHttpRequest postRequest = new HttpPost("/completion");
+    final String jsonRequest;
     try {
-      val json = JACKSON.writeValueAsString(request);
-      log.debug("Serialized request into JSON payload: {}", json);
-      postRequest.setEntity(new StringEntity(json, ContentType.APPLICATION_JSON));
+      jsonRequest = JACKSON.writeValueAsString(request);
+      log.debug("Serialized request into JSON payload: {}", jsonRequest);
     } catch (final JsonProcessingException e) {
       throw new OrchestrationClientException("Failed to serialize request parameters", e);
     }
 
-    return executeRequest(postRequest);
+    return executeRequest(jsonRequest);
+  }
+
+  /**
+   * Perform a request to the orchestration service using a module configuration provided as JSON
+   * string. This can be useful when building a configuration in the AI Launchpad UI and exporting
+   * it as JSON. Furthermore, this allows for using features that are not yet supported natively by
+   * the API.
+   *
+   * <p><b>NOTE:</b> This method does not support streaming.
+   *
+   * @param prompt The input parameters and optionally message history to use for prompt execution.
+   * @param moduleConfig The module configuration in JSON format.
+   * @return The completion response.
+   * @throws OrchestrationClientException If the request fails.
+   */
+  @Beta
+  @Nonnull
+  public OrchestrationChatResponse executeRequestFromJsonModuleConfig(
+      @Nonnull final OrchestrationPrompt prompt, @Nonnull final String moduleConfig)
+      throws OrchestrationClientException {
+    if (!prompt.getMessages().isEmpty()) {
+      throw new IllegalArgumentException(
+          "Prompt must not contain any messages when using a JSON module configuration, as the template is already defined in the JSON.");
+    }
+
+    final var request =
+        new CompletionPostRequest()
+            .messagesHistory(prompt.getMessagesHistory())
+            .inputParams(prompt.getTemplateParameters());
+
+    final ObjectNode requestJson = JACKSON.valueToTree(request);
+    final JsonNode moduleConfigJson;
+    try {
+      moduleConfigJson = JACKSON.readTree(moduleConfig);
+    } catch (JsonProcessingException e) {
+      throw new IllegalArgumentException(
+          "The provided module configuration is not valid JSON: " + moduleConfig, e);
+    }
+    requestJson.set("orchestration_config", moduleConfigJson);
```

Comment:
(Major/Question)

Wouldn't it make more sense applying the module config JSON to client instead of per request?
Seems like the only place where `orchestration_config` is provided, could be an optional parameter that is passed along the convenience API(?)

Reply:
Hm not exactly sure what you mean.. Do you mean setting the config on the `OrchestrationClient` object instead? Or something else?

Reply:
Yes.
Boiling this down to the question: Is `orchestration_config` to be expected **request specific** or **client specific**?

Reply:
Essentially same as the `OrchestrationConfig` object: It depends how much, if any, config changes per request. But the tendency is it is tied to the client, i.e. for typical use cases the majority of config will stay the same across multiple requests.

However, we decided to not allow config per client to make the API simpler. So the same strategy is applied for this method too. If we want to, we can still change this in the future though.

Reply:
To me, from a user perspective, it would feel weird to pass a `String` object along (over multiple controller classes potentially) just to pass to my request invocations.

But it's difficult to argue about client / request specific data handling, when I haven't yet seen a proper use-case "in the wild". Maybe we will discuss this again, after having users first interact with that.